src/js/contentscripts/dnt.js

```diff
@@ -20,28 +20,34 @@ function getPageScript() {
   // code below is not a content script: no chrome.* APIs /////////////////////
 
   // return a string
-  return "(" + function (NAVIGATOR, OBJECT) {
+  return "(" + function () {
 
     if (NAVIGATOR.doNotTrack != "1") {
       OBJECT.defineProperty(OBJECT.getPrototypeOf(NAVIGATOR), "doNotTrack", {
         get: function doNotTrack() {
           return "1";
-        }
+        },
+        configurable: true,
+        enumerable: true
       });
     }
 
-    if (!NAVIGATOR.globalPrivacyControl) {
+    if (!'globalPrivacyControl' in NAVIGATOR.prototype) {
       try {
-        OBJECT.defineProperty(NAVIGATOR, "globalPrivacyControl", {
-          value: true
+        OBJECT.defineProperty(NAVIGATOR.prototype, "globalPrivacyControl", {
+          get: function () {
+            return true;
+          },
+          configurable: true,
+          enumerable: true
         });
       } catch (e) {
         console.error("Privacy Badger failed to set navigator.globalPrivacyControl, probably because another extension set it in an incompatible way first.");
       }
     }
 
   // save locally to keep from getting overwritten by site code
-  } + "(window.navigator, Object));";
+  } + "(globalThis.Navigator, Object));";
```

Comment:
We should try to avoid `globalThis` as it's [not supported](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis#browser_compatibility) all the way back to our minimum browser versions ([Chrome 55](https://github.com/EFForg/privacybadger/blob/1493462cf4f0005d396d8c66fb8d2e64268b13b9/src/manifest.json#L503) and [Firefox 52](https://github.com/EFForg/privacybadger/blob/1493462cf4f0005d396d8c66fb8d2e64268b13b9/src/manifest.json#L9)).

Reply:
Content scripts only gets injected into contexts where `window` exists, so there is no need for `globalThis` anyway.

Reply:
Makes sense. I'll update this to use `self` or `window` 👍